### PR TITLE
TYP: mostly datetimelike

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/period.py [hunk 1]
```diff
@@ -421,7 +427,7 @@ def is_full(self) -> bool:
         """
         if len(self) == 0:
             return True
-        if not self.is_monotonic:
+        if not self.is_monotonic_increasing:
```
Comment (WillAyd) [inline pandas/core/indexes/period.py:430]: Does this have any behavioral changes?

Comment (jbrockmendel) [reply pandas/core/indexes/period.py:430]: No, just more explicit.

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/core/indexes/datetimelike.py [hunk 2]
```diff
@@ -157,6 +157,7 @@ def __contains__(self, key: Any) -> bool:
             res = self.get_loc(key)
         except (KeyError, TypeError, ValueError):
             return False
+        return True
```
Comment (jreback) [inline pandas/core/indexes/datetimelike.py:160]: what is this?

**Review by jreback — CHANGES_REQUESTED** [review]
question

File: pandas/core/indexes/datetimelike.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/indexes/datetimelike.py:160]: i think this and the FIXME weren't meant to be committed; will fix

Comment (jbrockmendel) [reply pandas/core/indexes/datetimelike.py]: updated + all-but-arm-green

**Comment by jreback** [comment]
looks good, is this compat with what we merged in : https://github.com/pandas-dev/pandas/pull/36742

maybe rebase just to be sure (and merge on green).

**Review by jreback — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**